qa-tests-backend/src/test/groovy/ImageSignaturesTest.groovy

```diff
@@ -0,0 +1,270 @@
+import static Services.waitForViolation
+
+import groups.Integration
+import io.stackrox.proto.storage.PolicyOuterClass
+import io.stackrox.proto.storage.PolicyOuterClass.Policy
+import io.stackrox.proto.storage.ScopeOuterClass
+import io.stackrox.proto.storage.SignatureIntegrationOuterClass.CosignPublicKeyVerification
+import io.stackrox.proto.storage.SignatureIntegrationOuterClass.SignatureIntegration
+import objects.Deployment
+import orchestratormanager.OrchestratorTypes
+import org.junit.experimental.categories.Category
+import services.PolicyService
+import services.SignatureIntegrationService
+import spock.lang.Shared
+import spock.lang.Unroll
+import util.Env
+
+class ImageSignaturesTest extends BaseSpecification {
+    // https://stack-rox.atlassian.net/browse/ROX-6891
+    static final private Integer WAIT_FOR_VIOLATION_TIMEOUT =
+            isRaceBuild() ? 450 : ((Env.mustGetOrchestratorType() == OrchestratorTypes.OPENSHIFT) ? 100 : 30)
+
+    static final private String SIGNATURE_TESTING_NAMESPACE = "qa-signature-tests"
+
+    // Names of the signature integration + policies that use the integration as the value of Trusted image signers.
+    static final private String DISTROLESS = "Distroless"
+    static final private String TEKTON = "Tekton"
+    static final private String UNVERIFIABLE = "Unverifiable"
+    // TODO(ROX-9996): Uncomment after we can handle multiple verification results.
+    //static final private String DISTROLESS_AND_TEKTON = "Distroless+Tekton"
+
+    // List of integration names used within tests.
+    // NOTE: If you add a new name, make sure to add it here.
+    static final private List<String> INTEGRATION_NAMES = [
+            DISTROLESS,
+            TEKTON,
+            UNVERIFIABLE,
+            // TODO(ROX-9996): Uncomment after we can handle multiple verification results.
+            //DISTROLESS_AND_TEKTON
+    ]
+
+    // Public keys used within signature integrations.
+    static final private Map<String, String> DISTROLESS_PUBLIC_KEY = [
+            // Source: https://raw.githubusercontent.com/GoogleContainerTools/distroless/main/cosign.pub
+            "Distroless": "-----BEGIN PUBLIC KEY-----\n" +
+                    "MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEWZzVzkb8A+DbgDpaJId/bOmV8n7Q\n" +
+                    "OqxYbK0Iro6GzSmOzxkn+N2AKawLyXi84WSwJQBK//psATakCgAQKkNTAA==\n" +
+                    "-----END PUBLIC KEY-----",
```

Comment:
nit: groovy supports multiline strings
```suggestion
            "Distroless": """-----BEGIN PUBLIC KEY-----
                    MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEWZzVzkb8A+DbgDpaJId/bOmV8n7Q
                    OqxYbK0Iro6GzSmOzxkn+N2AKawLyXi84WSwJQBK//psATakCgAQKkNTAA==
                    -----END PUBLIC KEY-----""".stripIndent(),
```

Reply:
Strip indent didn't work for me correctly, I had to resort to
```
"Distroless": """\
<public-key>
"""
```
instead.